read.go

```diff
@@ -159,28 +159,58 @@ func (r *reader) readHeader() ([]*Element, error) {
 		return nil, err
 	}
 
+	metaElems := []*Element{maybeMetaLen} // TODO: maybe set capacity to a reasonable initial size
+	metaElementGroupLengthDefined := true
 	if maybeMetaLen.Tag != tag.FileMetaInformationGroupLength || maybeMetaLen.Value.ValueType() != Ints {
-		return nil, ErrorMetaElementGroupLength
+		if !r.opts.allowErrorMetaElementGroupLength {
+			return nil, ErrorMetaElementGroupLength
+		}
+		metaElementGroupLengthDefined = false
+		metaElems = append(metaElems, maybeMetaLen)
 	}
 
-	metaLen := maybeMetaLen.Value.GetValue().([]int)[0]
-
-	metaElems := []*Element{maybeMetaLen} // TODO: maybe set capacity to a reasonable initial size
+	if metaElementGroupLengthDefined {
+		metaLen := maybeMetaLen.Value.GetValue().([]int)[0]
 
-	// Read the metadata elements
-	err = r.rawReader.PushLimit(int64(metaLen))
-	if err != nil {
-		return nil, err
-	}
-	defer r.rawReader.PopLimit()
-	for !r.rawReader.IsLimitExhausted() {
-		elem, err := r.readElement(nil, nil)
+		// Read the metadata elements
+		err = r.rawReader.PushLimit(int64(metaLen))
 		if err != nil {
-			// TODO: see if we can skip over malformed elements somehow
 			return nil, err
 		}
-		// log.Printf("Metadata Element: %s\n", elem)
-		metaElems = append(metaElems, elem)
+		defer r.rawReader.PopLimit()
+		for !r.rawReader.IsLimitExhausted() {
+			elem, err := r.readElement(nil, nil)
+			if err != nil {
+				// TODO: see if we can skip over malformed elements somehow
+				return nil, err
+			}
+			// log.Printf("Metadata Element: %s\n", elem)
+			metaElems = append(metaElems, elem)
+		}
+	} else if r.opts.allowErrorMetaElementGroupLength {
+		// We cannot use the limit functionality
+		debug.Log("Proceeding without metadata group length")
+		for {
+			// Lets peek into the tag field until we get to end-of-header
+			tag_bytes, err := r.rawReader.Peek(4)
+			if err != nil {
+				return nil, ErrorMetaElementGroupLength
+			}
+			tg := tag.Tag{}
+			buff := bytes.NewBuffer(tag_bytes)
+			binary.Read(buff, binary.LittleEndian, &tg)
```

Comment:
I'm not sure what would happen here if in the future an extra field were added to the tag struct in the future? It also sounds like unexported fields may lead to [Read panicing (if we were to add on in the future)](https://pkg.go.dev/encoding/binary#Read)

It's a bit more verbose, but we may want to consider something like
```
if err := binary.Read(buff, binary.LittleEndian, &tg.Group); err != nil {
// handle err
}
if err := binary.Read(buff, binary.LittleEndian, &tg.Element); err != nil {
//handle err
}
```

Also don't forget to handle the errors on binary.Read as well :). 

WDYT?

Reply:
Certainly I can change it.  I guess your point is that tag could change in a way that makes it incompatible with direct mapping from the DICOM header...

Reply:
Yep, structs can/do change and if it did, it could lead to an unintentional breakage here. If the tests are good enough it's possible that regression would be protected, but it seems safer to do the slightly more verbose way to me for now!

Reply:
Could we make this update? Thanks!

Reply:
Ah, not sure if this update was applied (it's not showing up for me). basically instead of reading into the tag.Tag directly, reading each field of the tag

```
if err := binary.Read(buff, binary.LittleEndian, &tg.Group); err != nil {
// handle err
}
if err := binary.Read(buff, binary.LittleEndian, &tg.Element); err != nil {
//handle err
}
```